File: backend/src/main/java/com/optimiza/clickbarber/model/Barbearia.java
package com.optimiza.clickbarber.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.*;

import java.util.List;
import java.util.UUID;

import static java.util.Objects.isNull;

@Table(name = "barbearias")
@Entity
@Getter
@Setter
@ToString
@EqualsAndHashCode
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class Barbearia {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "id_externo", unique = true, nullable = false)
    private UUID idExterno;

    private String cnpj;
    private String nome;
    private String endereco;
    private String telefone;

    @OneToOne
    @JoinColumn(name = "usuario_id")
    private Usuario usuario;

    @OneToMany(mappedBy = "barbearia", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<Servico> servicos;

    @OneToMany(mappedBy = "barbearia", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonIgnoreProperties("barbearia")
    private List<Barbeiro> barbeiros;

    @PrePersist
    void gerarIdExterno() {
        if (isNull(idExterno)) {
            idExterno = UUID.randomUUID();
        }
    }

}

File: frontend-admin/src/models/Barbearia.ts
import { Barbeiro } from "./Barbeiro";
import { HorarioFuncionamento } from "./HorarioFuncionamento";
import { Servico } from "./Servico";

export interface Barbearia {
    idExterno: string;
    cnpj: string;
    nome: string;
    email: string;
    endereco: string;
    telefone: string;
    servicos: Servico[];
    barbeiros: Barbeiro[];
    horarios: HorarioFuncionamento[];
}

export interface BarbeariaAtualizarDto {
    idExterno: string;
    cnpj: string;
    nome: string;
    email: string;
    endereco: string;
    telefone: string;
}